internal/k8s/controllers/gateway_class_config_controller.go

```diff
@@ -62,6 +64,16 @@ func (r *GatewayClassConfigReconciler) Reconcile(ctx context.Context, req ctrl.R
 	}
 
 	// we're creating or updating
+
+	// evict any class that's referencing an updated config from cache
+	if using, err := r.Client.GatewayClassesUsingConfig(ctx, gcc); err == nil && len(using.Items) > 0 {
+		for _, gc := range using.Items {
+			if err := r.Manager.DeleteGatewayClass(ctx, gc.Name); err != nil {
+				logger.Warn("error evicting cached gateway class referencing config", "error", err)
```

Comment:
@mikemorris @andrewstucki this is best effort right now. Should we keep it that way, or should we be erroring out if we potentially have a stale `GatewayClassConfig` in the cache?

Reply:
think we should be fine right now